GenericTrigger.lua

```diff
@@ -660,6 +660,9 @@ function GenericTrigger.Add(data, region)
               end
             end
           end
+          if (trigger.custom_type == "stateupdate") then
+            statesParameter = "full";
+          end
```

Comment:
Most of this patch just modifies the UI to not show certain elements for the new functionality. This line is one that matters.